### DISCUSSION TITLE
copy to clipboard?

### DISCUSSION THREAD

**Discussion by jesper-olsen** [helix-editor/helix#6847] [Q&A]
Copy to clipboard (⌘c) seems not to be implemented?
Paste (⌘v) is implemented...

Helix 23.03 on a Macbook M1 (Ventura 13.0)
<img width="1266" alt="Screenshot 2023-04-22 at 6 13 41" src="https://user-images.githubusercontent.com/43079279/233782008-631ddca2-b2bc-4ee2-8fda-3ca6c6b9dfae.png">

**Comment by David-Else** [comment]
What is your output for `hx --health clipboard`?

**Comment by jesper-olsen** [comment]
% hx --health clipboard

System clipboard provider: pbpaste+pbcopy

**Comment by jesper-olsen** [comment]
Turns out it is because of the mouse mode - has to be off for  copy to work...

[editor]
mouse = false

**Reply by kirawi** [reply]
That's strange, I have mouse mode enabled and it never caused me problems. I also don't think there would be any reason for it to happen looking at the code.

**Comment by jesper-olsen** [comment]
Are you on macOS like me?

vim and hx are the same for me - only difference is that mouse mode is off
by default for vim and on by default for hx. Mouse mode is nice - but not
at the cost of copy...

**Reply by kirawi** [reply]
Yes. I reread your post and I realized you weren't using the yank command. I'd recommend using `space-y` to copy instead which will allow you to keep mouse mode on.

**Reply by jesper-olsen** [reply]
Nice - どうもありがとう

**Reply by dawkrish** [reply]
isn't there something like
`clipboard=unnamedplus` for helix ?

**Comment by k4lizen** [comment]
> Yes. I reread your post and I realized you weren't using the yank command. I'd recommend using `space-y` to copy instead which will allow you to keep mouse mode on.

I ran into the same problem and came upon this issue. I often do mouse select -> copy. Currently only `space-y` works, and just `y` doesnt. `space-y` is a clunky shortcut for me to use for such a common action, is there a workaround?
Also is there a reason that `ctrl-shift-c` doesn't work?

**Reply by k4lizen** [reply]
@kirawi

**Reply by kirawi** [reply]
`y` yanks to a register, not the clipboard. You can rebind `y` to `"yank_main_selection_to_clipboard"` in your config to change that (though you'll likely find registers are more often used than the clipboard once you learn how to use them). What are you expecting C-C to do?

**Reply by k4lizen** [reply]
> You can rebind y to "yank_main_selection_to_clipboard" in your config to change that

Thanks!
> though you'll likely find registers are more often used than the clipboard once you learn how to use them

I don't use helix as my editor (for now) (I use vscode(ium) for that), just use it when I need to quickly edit/skim some file, which is commonly accompanied with some copy-pasting action. I'm not really proficient in helix, but I'm assuming I can't use the registers if I want to copy something to out of helix right?
> What are you expecting C-C to do?

`ctrl-shift-c` is "copy to clipboard" on most terminals.

**Reply by kirawi** [reply]
I don't think any TUI editor follows the convention of using C-C for yanking. Among other reasons, I'm guessing it's also because it conflicts with the terminal binding.

**Reply by pascalkuthe** [reply]
Helix is a vim/kakoune like editor. Mouse support is not a priority and behaving like vscode or other guis is a non-goal.

**Reply by k4lizen** [reply]
> I don't think any TUI editor follows the convention of using C-C for yanking. Among other reasons, I'm guessing it's also because it conflicts with the terminal binding.

It doesn't conflict from my experience at least, it simply doesn't work
> Mouse support is not a priority

Yeah but its nice to have
> and behaving like vscode or other guis is a non-goal.

Thats understandable

As a side-note, tmux in vim-mode (and probably emacs as well) copies to clipboard on mouse select (though I understand thats apples/oranges somewhat, its good qol on their part (since they also support regular vim visual mode selection and yanking))

**Reply by HernandoR** [reply]
based on my understanding of hx, Should we beable to bind (or mirror) register `+` and `"` ? Such config should provide a method to mirror the default register and the clipboard register.

**Reply by volt-l18** [reply]
> onflict from my experience at least, it simply doesn't work

using u for undo was kinda clunky for me at being as well at first.

**Comment by markstos** [comment]
On some terminals if you hold down "shift" and select text, I believe the selection belongs to the terminal instead of the app, and the terminal itself may be configured to copy its selection to the system clipboard. I believe this works on wezterm, foot and alacritty. I've tested it today on Wezterm. I recall it working on `foot` and I _think_ it worked on Alacritty.

**Reply by k4lizen** [reply]
Ou it works for me on kitty! This is the best solution imo. Thank you!

**Comment by xtrasmal** [comment]
I'm on Alacritty. 
I press 'v', then make a selection. 
Then press `space-Y`
Then you can paste it everywhere you want.

**Reply by k4lizen** [reply]
Neat! Selecting with mouse in normal mode and doing `space-Y` works for me, but the shortcut is quite cumbersome to do for something such as copying which I do often (hard to do with left hand on qwerty, but I use colemak-dh which makes it impossible so I have to move my hand right hand off my mouse to the keyboard and back, which is insanely bad)

**Reply by achkasov** [reply]
@k4lizen 
On sway + kitty + helix:

1. Select text in V-mode,
2. space-y to copy.
3. ctrl+shift+v to paste in any other kitty

**Comment by lineality** [comment]
Many thanks for discussion. 
I am using Ubuntu (usually I am on Fedora) and space-y or space-Y in v mode or otherwise does not work to send highlighed text to system clipboard (I have not disabled mouse). --version helix 24.7 (db1d8425), gnome terminal Version 3.52.0 for GNOME 46
(Though it seems like copy-paste goes haywire on all systems over the last year or so for everyone I know... I thought that was highly stable historically. Is new clipboard security/sandboxing (or lack of) wrenching everything up?)
Helix is amazing, many thanks.

**Reply by HernandoR** [reply]
To my experience in Ubuntu and kubuntu you may need to check the following
1. If helix was installed in snap
2. If x-clip is installed

I'm not sure if the latter one is actually required, but you could have a try.

**Comment by denppa** [comment]
For those that are looking for the behavior of yanking a selection into the system clipboard without rebinding the yank keys, or using the <space+y> key chord, this setting can be used:

```toml
[editor]
default-yank-register = "+" 
```

In a sense, this is similar to what `clipboard=unnamedplus` does in neovim.
